### PR TITLE
make usable jar

### PR DESCRIPTION
**Pull request by akiezun**
tested on mac by using local maven to `install`. I tested gatk (on mac) by adding the hdf5 dependency and running tests from inside the gatk5 repo.

notice that i've managed to test this repo on linux and mac on travis!

### CODE REVIEW

**Comment by akiezun** [comment]
@lbergelson can you have a look?

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+defaultTasks 'jar'
+
+//Delete the windows script - we never test on Windows so let's not pretend it works
+startScripts {
+    doLast {
+        delete windowsScript
+    }
+}
+
+task unzipHDF5JAR(type: Copy) {
```
Comment (lbergelson) [inline build.gradle]: could you add a comment here about what this is doing?

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+dependencies {
+    compile 'org.apache.commons:commons-lang3:3.4'
+    compile 'org.apache.commons:commons-math3:3.5'
+    compile 'commons-io:commons-io:2.4'
+    compile 'org.broadinstitute:gatk-native-bindings:0.0.3'
+
+    testCompile 'org.testng:testng:6.9.6'
+    compile 'org.apache.hadoop:hadoop-minicluster:2.7.2' //the version of minicluster should match the version of hadoop
+
+    compileOnly name: 'jarhdf5-2.11.0'
```
Comment (lbergelson) [inline build.gradle]: could you add a comment about why you're doing this here

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+
+dependencies {
+    compile 'org.apache.commons:commons-lang3:3.4'
+    compile 'org.apache.commons:commons-math3:3.5'
+    compile 'commons-io:commons-io:2.4'
+    compile 'org.broadinstitute:gatk-native-bindings:0.0.3'
+
+    testCompile 'org.testng:testng:6.9.6'
+    compile 'org.apache.hadoop:hadoop-minicluster:2.7.2' //the version of minicluster should match the version of hadoop
```
Comment (lbergelson) [inline build.gradle]: it would be good if we could avoid including this

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+buildscript {
+    repositories {
+        mavenCentral()
+        jcenter() // for shadow plugin
+     }
+}
+
+plugins {
+    id "java"
+    id "application"
```
Comment (lbergelson) [inline build.gradle]: no need for application plugin

File: build.gradle (hunk 3 above)
Comment (lbergelson) [inline build.gradle]: you can replace this with these loggers:

```
    compile 'org.apache.logging.log4j:log4j-api:2.3'
    compile 'org.apache.logging.log4j:log4j-core:2.3'
    compile 'org.slf4j:slf4j-api:1.7.21'
```

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    compile 'commons-io:commons-io:2.4'
+    compile 'org.broadinstitute:gatk-native-bindings:0.0.3'
+
+    testCompile 'org.testng:testng:6.9.6'
+    compile 'org.apache.hadoop:hadoop-minicluster:2.7.2' //the version of minicluster should match the version of hadoop
+
+    compileOnly name: 'jarhdf5-2.11.0'
+}
+
+mainClassName = "org.broadinstitute.hdf5.HDF5Library"
```
Comment (lbergelson) [inline build.gradle]: remove this one application is removed

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    String CI = "$System.env.CI"
+    useTestNG{
+
+    }
+
+    // set heap size for the test JVM(s)
+    minHeapSize = "1G"
+    maxHeapSize = "2G"
+
+    if (CI == "true") {
```
Comment (lbergelson) [inline build.gradle]: you can remove this condition, the else is all this project needs

File: src/main/java/org/broadinstitute/hdf5/HDF5Library.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+ *
+ * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
+ */
+public final class HDF5Library implements NativeLibrary {
+    private static final Logger logger = LogManager.getLogger(HDF5Library.class);
+
+    private static boolean loaded = false;
+    public static final String HDF5_LIB_NAME = "jhdf5.2.11.0";
+
+    public static void main(String[] args){
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hdf5/HDF5Library.java]: kill this main

File: build.gradle [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+version = "0.1.0"
+//(isRelease ? gitVersion() : gitVersion() + "-SNAPSHOT").replaceAll(".dirty", "")
+
+logger.info("build for version:" + version)
+group = 'org.broadinstitute'
+
+defaultTasks 'jar'
+
+//Delete the windows script - we never test on Windows so let's not pretend it works
+startScripts {
```
Comment (lbergelson) [inline build.gradle]: don't need this block

File: build.gradle [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+    archives sourcesJar
+}
+
+/**
+ * Sign non-snapshot releases with our secret key.  This should never need to be invoked directly.
+ */
+signing {
+    required { isRelease && gradle.taskGraph.hasTask("uploadArchives") }
+    sign configurations.archives
+}
```
Comment (lbergelson) [inline build.gradle:182]: Could you add an uploadArchives task?  The one in htsjdk is is a good one to model it on.

**Comment by lbergelson** [comment]
@akiezun Back to you, looks good, just drop the application specific stuff and minicluster.

File: build.gradle (hunk 1 above)
Comment (akiezun) [inline build.gradle]: done

File: build.gradle (hunk 2 above)
Comment (akiezun) [inline build.gradle]: done

File: build.gradle (hunk 3 above)
Comment (akiezun) [inline build.gradle]: does not work for me. On `./gradlew clean compileJava` I get

```
/Users/akiezun/IdeaProjects/hdf5-java-bindings/src/main/java/org/broadinstitute/hdf5/HDF5Library.java:7: error: package org.apache.log4j does not exist
import org.apache.log4j.LogManager;
                       ^
```

File: build.gradle (hunk 4 above)
Comment (akiezun) [inline build.gradle]: done

File: build.gradle (hunk 5 above)
Comment (akiezun) [inline build.gradle]: done

File: build.gradle (hunk 6 above)
Comment (akiezun) [inline build.gradle]: done

File: src/main/java/org/broadinstitute/hdf5/HDF5Library.java (hunk 7 above)
Comment (akiezun) [inline src/main/java/org/broadinstitute/hdf5/HDF5Library.java]: done

File: build.gradle (hunk 8 above)
Comment (akiezun) [inline build.gradle]: fixed

File: build.gradle (hunk 3 above)
Comment (akiezun) [inline build.gradle]: I put this in and now it seems to work 

```
compile 'org.slf4j:slf4j-log4j12:1.7.21'
```

File: build.gradle (hunk 9 above)
Comment (akiezun) [inline build.gradle:182]: done

**Pull request merged**